js/ui/tab_panel.js

```diff
@@ -317,13 +317,14 @@ var TabPanel = MultiView.inherit({
             loopItemFocus: this.option("loop"),
             selectionRequired: true,
             onOptionChanged: (function(args) {
-                var name = args.name,
-                    value = $(args.value);
-
-                if(name === "focusedElement") {
-                    var id = value ? value.index() : value;
-                    var newItem = value ? this._itemElements().eq(id) : value;
-                    this.option("focusedElement", getPublicElement(newItem));
+                if(args.name === "focusedElement") {
+                    if(args.value) {
```

Comment:
Maybe it would be better to extract method?

```
if(args.name === "focusedElement") {
  this.option("focusedElement", this._getItemElement(args.value));
}

Reply:
As for me, it is unnecessary:
Just 6 lines.
Easy to read, as it is documented by variables.
Unable to reuse.

But I can suggest the following refactoring:
```JavaScript
                    if(args.value) {
                        var newItemIndex = $(args.value).index();
                        var $newItem = this._itemElements().eq(newItemIndex);
                        this.option("focusedElement", getPublicElement($newItem));
                    } else {
                        this.option("focusedElement", args.value);
                    }
```

Reply:
LGTM